### ISSUE TITLE
Swiper goes back on slide 1 on key press

### ISSUE THREAD

**Issue by alxpereira** [leecade/react-native-swiper#839]
### Version
Which versions are you using: OS-X High Sierre / iOS 11

- react-native-swiper v1.5.*
- react-native v0.55.2

### Expected behaviour

When i'm writing in a Form (Input), it should stay on the same slide.

### Actual behaviour

When I'm writing in a Form (input), the swiper shows back the first slide on keypress (after the keyboard shows up). When I click back on the screen it goes back on the 2nd slide.

### Steps to reproduce
1. Create a swiper with 2 or 3 slides
2. Put in the swiper slide 2 an input
3. When editing this input (on simulator) you'll see the swiper going back to slide 1.

**Comment by anasalsaadi14** [comment]
any solutions ?

**Comment by Aryk** [comment]
yeah, I'm getting the same issue. Trying to find a work around....

**Comment by Aryk** [comment]
When I put loop=false it does not happen, but then I lose that functionality.

**Comment by Aryk** [comment]
Hey guys, have a few work arounds:

1) Put loop=false, 
2) Wrap the Swiper with memo and make it only change when the index or anything inside it changes. The state changing and re-rendering is what is causing this glitch.

Both of these fixed it for me.

**Comment by aleyango** [comment]
any update on this?

**Comment by nguyenanhthai** [comment]
Same issue. Any solution?

**Comment by dattran1232003** [comment]
Here is the issue when I focus on a TextInput
![image](https://github.com/leecade/react-native-swiper/assets/57513789/b1f30543-d214-4ab5-9749-08c21cf5dc2b)

----------
It won't happen if I remove these lines of code, and remove the usage of `isKeyboardVisible` variable
```typescript
  const [isKeyboardVisible, setIsKeyboardVisible] = useState(false)
  useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      () => {
        setIsKeyboardVisible(true)
      },
    )

    const keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      () => {
        setIsKeyboardVisible(false)
      },
    )

    // Clean up the listeners when the component unmounts
    return () => {
      keyboardDidShowListener.remove()
      keyboardDidHideListener.remove()
    }
  })
```
